addons/payment_authorize/models/payment_transaction.py

```diff
@@ -95,18 +95,37 @@ def _send_refund_request(self, amount_to_refund=None, create_refund_transaction=
                 create_refund_transaction=create_refund_transaction,
             )
 
-        refund_tx = super()._send_refund_request(
-            amount_to_refund=amount_to_refund, create_refund_transaction=False
-        )
+        # Only full refund is possible.
 
         authorize_API = AuthorizeAPI(self.acquirer_id)
-        rounded_amount = round(self.amount, self.currency_id.decimal_places)
-        res_content = authorize_API.refund(self.acquirer_reference, rounded_amount)
+        tx_details = authorize_API.get_transaction_details(self.acquirer_reference)
+
+        if tx_details and tx_details.get('err_code'):
+            res_content = {
+                'x_response_code': AuthorizeAPI.AUTH_ERROR_STATUS,
+                'x_response_reason_text': tx_details.get('err_msg')
+            }
+
+            refund_tx = super()._send_refund_request(
+                amount_to_refund=amount_to_refund, create_refund_transaction=False
+            )
+        elif tx_details.get('transaction', {}).get('transactionStatus') in ['capturedPendingSettlement', 'authorizedPendingCapture']:
+            refund_tx = super()._send_refund_request(
+                amount_to_refund=self.amount, create_refund_transaction=True
+            )
+            res_content = authorize_API.void(self.acquirer_reference)
+        else:
+            refund_tx = super()._send_refund_request(
+                amount_to_refund=amount_to_refund, create_refund_transaction=True
+            )
```

Comment:
Same here, you can add a comment with something like:
"Pending tx can't be refund, they must be void."

Reply:
done :)